spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/statsd/StatsdMetricsExportAutoConfiguration.java

```diff
@@ -59,4 +60,8 @@ public StatsdMeterRegistry statsdMeterRegistry(StatsdConfig statsdConfig, Clock
 		return new StatsdMeterRegistry(statsdConfig, clock);
 	}
 
+	@PreDestroy
+	public void destroy(StatsdMeterRegistry statsdMeterRegistry) {
+		statsdMeterRegistry.close();
+	}
 }
```

Comment:
This won't work as `@PrreDestroy` should not have any arguments (See https://jakarta.ee/specifications/platform/9/apidocs/jakarta/annotation/predestroy). 

Finally as the `StatsdMeterRegistry` has a [`close`](https://www.javadoc.io/doc/io.micrometer/micrometer-registry-statsd/latest/io/micrometer/statsd/StatsdMeterRegistry.html#close()) method that should already be registered automatically as a pre-destroy method. For this see the [`@Bean`](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Bean.html#destroyMethod()). 

So the solution to the problem lies not in adding additional annotations but probably elsewhere.

Reply:
Thanks, @mdeinum. I was just going to comment with the same analysis.

@popotec can you please open an issue so that we can investigate further? Seeing the stack trace of the exception when bean creation is attempted during shutdown may be enough for us to figure out the cause.